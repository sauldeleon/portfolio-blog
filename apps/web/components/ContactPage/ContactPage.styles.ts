import { styled } from 'styled-components'

import { MainPortal } from '../MainPortal/MainPortal'

export const StyledMainPortal = styled(MainPortal)`
  margin-top: 166px;
`
