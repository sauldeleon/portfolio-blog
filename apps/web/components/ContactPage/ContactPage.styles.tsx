import Image from 'next/image'
import styled, { css, keyframes } from 'styled-components'

import { Body, Heading } from '@sdlgr/typography'

import { MainPortal } from '@web/components/MainPortal/MainPortal'
import { randomIntFromInterval } from '@web/utils/random'

const TEETH_DELAY_FACTOR = 15

/**                           
 *   
  Each 2 values, it doubles the previous increment
  t(0) = 5*1 + 0 = 5
  t(1) = 5*1 + t(n-1) = 10 
  t(2) = 5*2 + t(n-1) = 20
  t(3) = 5*2 + t(n-1) = 30
  t(4) = 5*4 + t(n-1) = 50
  t(5) = 5*4 + t(n-1) = 70
  t(6) = 5*8 + t(n-1) = 110
  t(7) = 5*8 + t(n-1) = 150

  t(0) = 5
  t(n) = t(n - 1) + 5 * 2^(n // 2)
*/
const getDelay = (n: number): number => {
  if (n === 0) {
    return TEETH_DELAY_FACTOR
  } else {
    return getDelay(n - 1) + TEETH_DELAY_FACTOR * Math.pow(2, Math.floor(n / 2))
  }
}

export const StyledMainPortal = styled(MainPortal)`
  margin-top: 30px;

  ${({ theme }) => theme.media.up.md} {
    margin-top: 166px;
  }
`

export const StyledContactInfoWrapper = styled.div`
  width: 100%;
  height: 100%;
`

export const StyledContactInfo = styled.div<{ $isVisible: boolean }>`
  display: grid;
  place-items: center;
  rotate: 270deg;
  position: absolute;
  top: 36px;
  left: 65px;
  opacity: 0;
  transition: opacity 2s ease-in;
  transform-origin: 50% 51%;

  ${({ $isVisible }) =>
    $isVisible &&
    css`
      opacity: 1;
    `}

  ${({ theme }) => theme.media.up.md} {
    position: unset;
    top: unset;
    left: unset;
    rotate: 0deg;
    row-gap: 10px;
  }
`

export const StyledHeading = styled(Heading)`
  ${({ theme }) => theme.typography.body.M};

  ${({ theme }) => theme.media.up.md} {
    ${({ theme }) => theme.typography.heading.heading2};
  }
`

export const StyledLabel = styled.span`
  display: inline-block;

  ${({ theme }) => theme.media.down.md} {
    display: none;
  }
`

export const StyledBody = styled(Body)`
  ${({ theme }) => theme.typography.body.XS};

  ${({ theme }) => theme.media.up.md} {
    ${({ theme }) => theme.typography.body.M};
  }
`

export const StyledPortrait = styled(Image)`
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;

  ${({ theme }) => theme.media.up.md} {
    width: 140px;
    height: 140px;
  }
`

export const PortraitContainer = styled.div``

interface StyledToothProps {
  $p1h?: number
  $p1v?: number
  $p2h?: number
  $p2v?: number
  $cp1h?: number
  $cp1v?: number
  $cp2h?: number
  $cp2v?: number
  $index: number
}

export const StyledTooth = styled.div<StyledToothProps>`
  ${({
    $p1h = 0,
    $p1v = 0,
    $p2h = -558,
    $p2v = 220,
    $cp1h = 0,
    $cp1v = 250,
    $cp2h = -280,
    $cp2v = 200,
    $index,
    theme,
  }) => css`
    position: absolute;
    width: 100%;
    height: 100%;
    top: -200%;
    left: -6px;
    animation-name: ${theme.animation.moveToothFall};
    animation-duration: 5s;
    animation-delay: ${getDelay($index)}s;
    animation-iteration-count: 1;

    ${theme.media.up.md} {
      --p1h: ${$p1h}; //initial x position
      --p1v: ${$p1v}; //initial y position
      --p2h: ${$p2h}; //final x position
      --p2v: ${$p2v}; //final y position
      --cp1h: ${$cp1h}; //initial x control curve position
      --cp1v: ${$cp1v}; //initial y control curve position
      --cp2h: ${$cp2h}; //final x control curve position
      --cp2v: ${$cp2v}; //final y control curve position
      --final-tooth-top: ${randomIntFromInterval(
        10,
        90,
        `tooth-height-${$index}`
      )}%;

      top: 0;
      left: -70%;
      animation-name: ${theme.animation.moveToothHorizontal},
        ${theme.animation.moveToothVertical};
      animation-timing-function: cubic-bezier(
          calc(1 / 3),
          calc((var(--cp1h) - var(--p1h)) / (var(--p2h) - var(--p1h))),
          calc(2 / 3),
          calc((var(--cp2h) - var(--p1h)) / (var(--p2h) - var(--p1h)))
        ),
        cubic-bezier(
          calc(1 / 3),
          calc((var(--cp1v) - var(--p1v)) / (var(--p2v) - var(--p1v))),
          calc(2 / 3),
          calc((var(--cp2v) - var(--p1v)) / (var(--p2v) - var(--p1v)))
        );
    }
  `}
`

export const RotateTooth = styled.div`
  position: absolute;
  animation: 2s ${({ theme }) => theme.animation.moveToothRotate} 1s linear
    infinite;
  top: 50px;
  left: 49.1%;
  transform: rotate(180deg);
`

export const StyledPortraitContainer = styled.div`
  position: relative;
  z-index: -1;
`

const getWeightedDelay = (delay: number, totalDelay: number) =>
  (delay * 100) / totalDelay

const getAdditiveBackgroundUrl = (images: string[], index: number) =>
  images
    .slice(0, index + 1)
    .map((image) => `url(${image})`)
    .join(', ')

const swapImageAnimation = (
  images: string[],
  totalImagesDelay: number
) => keyframes`
      0% {
        background-image: none;
      }
      ${images.map((_, index) => {
        const weightedDelay = getWeightedDelay(
          getDelay(index),
          totalImagesDelay
        )
        return css`
          ${weightedDelay - 0.0001}% {
            background-image: ${index > 0
              ? getAdditiveBackgroundUrl(images, index - 1)
              : 'none'};
          }

          ${weightedDelay}% {
            background-image: ${getAdditiveBackgroundUrl(images, index)};
          }
        `
      })}
`

export const ToothHoleImage = styled.div<{
  $images: string[]
}>`
  position: absolute;
  background-image: none;
  background-size: contain;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  ${({ $images }) => {
    const totalImagesDelay = getDelay($images.length - 1)

    return css`
      animation: ${totalImagesDelay}s
        ${swapImageAnimation($images, totalImagesDelay)} linear forwards;
    `
  }}
`
