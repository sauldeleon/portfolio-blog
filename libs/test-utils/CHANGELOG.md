# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
