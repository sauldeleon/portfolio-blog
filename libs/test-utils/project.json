{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
