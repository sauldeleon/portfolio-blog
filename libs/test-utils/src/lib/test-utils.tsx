import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
