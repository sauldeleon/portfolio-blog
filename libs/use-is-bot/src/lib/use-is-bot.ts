import { load } from '@fingerprintjs/botd'
import { useEffect, useState } from 'react'

export interface UseIsBotProps {
  afterDetection?: () => void
}

export function useIsBot(props?: UseIsBotProps) {
  const { afterDetection } = props ?? {}
  const [isBot, setIsBot] = useState(true)
  const [isLoading, setIsLoading] = useState(true)

  useEffect(() => {
    const botdPromise = load()
    botdPromise
      .then((botd) => botd.detect())
      .then((result) => {
        setIsBot(result.bot)
        setIsLoading(false)
        afterDetection?.()
      })
  }, [afterDetection])

  return { isBot, isLoading }
}
