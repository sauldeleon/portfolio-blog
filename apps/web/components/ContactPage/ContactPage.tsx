'use client'

import { useState } from 'react'

import { Link } from '@sdlgr/link'
import { useIsBot } from '@sdlgr/use-is-bot'

import { MainPortal } from '@web/components/MainPortal/MainPortal'
import { NoSSR } from '@web/components/NoSSR/NoSSR'
import { useClientTranslation } from '@web/i18n/client'

import {
  StyledBody,
  StyledContactInfo,
  StyledContactInfoWrapper,
  StyledHeading,
  StyledLabel,
} from './ContactPage.styles'
import { Portrait } from './components/Portrait/Portrait'
import { ToothLessPortrait } from './components/ToothLessPortrait/ToothLessPortrait'

export function ContactPage() {
  const { t } = useClientTranslation('contactPage')
  const [painMode, setPainMode] = useState(false)

  const { isBot, isLoading } = useIsBot()

  const mail = 'sauldeleonguerrero@gmail.com'

  return (
    <MainPortal>
      <StyledContactInfoWrapper>
        <StyledContactInfo $isVisible={!isLoading} data-testid="contact-info">
          <NoSSR>
            {painMode ? (
              <ToothLessPortrait onClick={() => setPainMode(false)} />
            ) : (
              <Portrait onClick={() => setPainMode(true)} />
            )}
          </NoSSR>
          <StyledHeading $level={2}>Software Engineer</StyledHeading>
          <Link
            href="https://www.linkedin.com/in/sauldeleonguerrero"
            aria-label={t('linkedInAria')}
          >
            <StyledBody $level="L">Linked in</StyledBody>
          </Link>
          {!isBot && (
            <NoSSR>
              <StyledBody $level="L">
                <StyledLabel>{t('phone')}:</StyledLabel>{' '}
                <Link href="tel:+34661030038" aria-label={t('phoneAria')}>
                  (+34) 661 03 00 38
                </Link>
              </StyledBody>
              <StyledBody $level="L">
                <StyledLabel>{t('mail')}:</StyledLabel>{' '}
                <Link href={`mailto:${mail}`} aria-label={t('emailAria')}>
                  {mail}
                </Link>
              </StyledBody>
            </NoSSR>
          )}
        </StyledContactInfo>
      </StyledContactInfoWrapper>
    </MainPortal>
  )
}
