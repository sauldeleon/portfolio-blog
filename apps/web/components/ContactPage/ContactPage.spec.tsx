import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { renderApp } from '@sdlgr/test-utils'
import { useIsBot } from '@sdlgr/use-is-bot'

import { ContactPage } from './ContactPage'

jest.mock('@sdlgr/use-is-bot', () => ({
  useIsBot: jest.fn(),
}))

describe('ContactPage', () => {
  it('should render successfully', async () => {
    ;(useIsBot as jest.Mock).mockReturnValue({
      isBot: false,
      isLoading: false,
    })
    const { baseElement } = renderApp(<ContactPage />)
    await screen.findByText('Software Engineer')
    expect(baseElement).toMatchSnapshot()
  })

  it('should swap images onClick', async () => {
    ;(useIsBot as jest.Mock).mockReturnValue({
      isBot: false,
      isLoading: false,
    })
    renderApp(<ContactPage />)
    await screen.findByText('Software Engineer')

    await userEvent.click(screen.getAllByAltText(/My profile picture/)[0])
    expect(
      await screen.findByAltText('My toothless profile picture'),
    ).toBeInTheDocument()
    await userEvent.click(screen.getByAltText(/My toothless profile picture/))
    expect(await screen.findAllByAltText(/My profile picture/)).toHaveLength(6)
  })

  it('should hide contact info if bot info is loading', async () => {
    ;(useIsBot as jest.Mock).mockReturnValue({
      isBot: false,
      isLoading: true,
    })
    renderApp(<ContactPage />)
    await screen.findByText('Software Engineer')
    expect(screen.getByTestId('contact-info')).toHaveStyleRule('opacity', '0')
  })
})
