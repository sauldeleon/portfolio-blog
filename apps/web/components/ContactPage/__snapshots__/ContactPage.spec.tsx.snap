// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage should render successfully 1`] = `
.c3 {
  z-index: 2;
  width: 180%;
  top: 14%;
  left: -41%;
  position: absolute;
  display: flex;
  justify-content: space-between;
  transform: rotate(90deg);
}

.c5 {
  width: 130px;
  height: 249px;
}

.c4 {
  background-color: #000000;
  border-top-right-radius: 42px 50%;
  border-bottom-right-radius: 44px 50%;
  border-top-left-radius: 44px 50%;
  border-bottom-left-radius: 44px 50%;
  margin-bottom: 5px;
  line-height: 0px;
}

.c6 {
  border-top-right-radius: 41px 50%;
  border-bottom-right-radius: 44px 50%;
  border-top-left-radius: 43px 50%;
  border-bottom-left-radius: 45px 50%;
  background-color: #000000;
  margin-bottom: 5px;
  line-height: 0px;
  transform: scaleX(-1);
}

.c7 {
  --height: 212px;
  position: absolute;
  overflow: clip;
  margin: 30px auto;
  height: 212px;
  border-top-left-radius: 30px 50%;
  border-bottom-left-radius: 30px 50%;
  border-top-right-radius: 31px 50%;
  border-bottom-right-radius: 32px 50%;
  top: -12px;
  left: 62px;
  right: 61px;
}

.c0 {
  --transform-origin-first: bottom;
  --transform-origin-last: top;
  --perspective: 100px;
  --rotate3D-first: 1,0,0,-45deg;
  --rotate3D-last: 1,0,0,45deg;
  --height-middle: 60%;
  --width-middle: unset;
  position: relative;
  display: flex;
  flex-direction: column;
  aspect-ratio: 1/2;
  align-items: stretch;
  gap: 15px;
  width: 100%;
  max-width: 300px;
}

.c1 {
  flex-grow: 1;
  z-index: -1;
  border: 1px solid #FFDD83;
  border-top: none;
  transform-origin: var(--transform-origin-first);
  transform: perspective(var(--perspective)) rotate3D(var(--rotate3D-first));
  background-color: #000000;
}

.c2 {
  position: relative;
  border: 1px solid;
  border-image-slice: 1;
  border-image-source: linear-gradient(           to bottom,           #FFDD83,           #98DFD6         );
  height: var(--height-middle);
  width: var(--width-middle);
  background-color: #000000;
}

.c23 {
  z-index: -1;
  flex-grow: 1;
  border: 1px solid #98DFD6;
  border-bottom: none;
  transform-origin: var(--transform-origin-last);
  transform: perspective(var(--perspective)) rotate3D(var(--rotate3D-last));
  background-color: #000000;
}

.c22 {
  position: relative;
  position: relative;
}

.c22::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: currentColor;
}

.c22::after {
  transform: scaleX(0);
  transform-origin: bottom right;
  transition: transform 0.25s ease-out;
}

.c22:hover::after {
  transform: scaleX(1);
  transform-origin: bottom left;
}

.c22:hover,
.c22:active,
.c22:focus {
  text-decoration: none;
}

.c22:focus-visible {
  outline: #FBFBFB auto 1px;
}

.c17 {
  font-family: fontFamily;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 1.5rem;
  line-height: normal;
  margin: 0;
}

.c19 {
  font-family: fontFamily;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 1rem;
  line-height: normal;
  white-space: pre-line;
}

.c19 ul,
.c19 ol {
  margin-left: 40px;
  margin-bottom: 10px;
}

.c19 ul {
  list-style: outside disc;
}

.c19 ol {
  list-style: decimal;
}

.c19 li {
  padding-left: 10px;
}

.c8 {
  width: 100%;
  height: 100%;
  display: grid;
  place-items: center;
}

.c9 {
  display: grid;
  place-items: center;
  rotate: 270deg;
  opacity: 1;
  transition: opacity 2s ease-in;
  transform-origin: 50% 51.7%;
  margin-bottom: 5px;
  row-gap: 4px;
}

.c18 {
  font-family: fontFamily;
  font-weight: 400;
  color: #FBFBFB;
  font-size: 0.9rem;
  line-height: normal;
}

.c20 {
  display: flex;
  align-items: center;
  font-family: fontFamily;
  font-weight: 500;
  color: #FBFBFB;
  font-size: 0.55rem;
  line-height: normal;
}

.c21 {
  display: grid;
  place-items: center;
  margin-right: 4px;
}

.c21 svg {
  width: 12px;
  height: 12px;
}

.c10 {
  position: relative;
  width: 70px;
  height: 70px;
}

.c11 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  position: absolute;
  opacity: 1;
}

.c12 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  position: absolute;
  opacity: 0;
  animation: BLSUr 60s 10s infinite;
}

.c13 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  position: absolute;
  opacity: 0;
  animation: feYcwF 60s 10s infinite;
}

.c14 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  position: absolute;
  opacity: 0;
  animation: gBIqQN 60s 10s infinite;
}

.c15 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  position: absolute;
  opacity: 0;
  animation: jCGkWp 60s 10s infinite;
}

.c16 {
  overflow: hidden;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  position: absolute;
  opacity: 0;
  animation: cJztmO 60s 10s infinite;
}

@media (width >= 768px) {
  .c3 {
    width: 152%;
    top: unset;
    left: -26%;
    margin: 18px auto;
    transform: unset;
  }
}

@media (width >= 768px) {
  .c5 {
    width: unset;
    height: unset;
  }
}

@media (width >= 768px) {
  .c4 {
    background-color: #000000;
    border-top-right-radius: 59px 50%;
    border-bottom-right-radius: 61px 50%;
    border-top-left-radius: 62px 50%;
    border-bottom-left-radius: 62px 50%;
  }
}

@media (width >= 768px) {
  .c6 {
    border-top-right-radius: 59px 50%;
    border-bottom-right-radius: 62px 50%;
    border-top-left-radius: 61px 50%;
    border-bottom-left-radius: 63px 50%;
  }
}

@media (width >= 768px) {
  .c7 {
    height: 305px;
    border-top-left-radius: 45px 50%;
    border-bottom-left-radius: 45px 50%;
    border-top-right-radius: 40px 50%;
    border-bottom-right-radius: 40px 50%;
    top: -4px;
    left: 91px;
    right: 91px;
  }
}

@media (width >= 768px) {
  .c0 {
    --transform-origin-first: right;
    --transform-origin-last: left;
    --perspective: 300px;
    --rotate3D-first: 0,1,0,45deg;
    --rotate3D-last: 0,1,0,-45deg;
    --height-middle: unset;
    --width-middle: clamp(0px, 60%, 830px);
    flex-direction: row;
    aspect-ratio: unset;
    height: 395px;
    overflow-x: clip;
    max-width: unset;
  }
}

@media (width >= 1024px) {
  .c0 {
    --perspective: 400px;
  }
}

@media (width >= 1440px) {
  .c0 {
    --perspective: 600px;
  }
}

@media (width >= 768px) {
  .c1 {
    border-top: 1px solid #FFDD83;
    border-left: none;
  }
}

@media (width >= 768px) {
  .c2 {
    border-image-source: linear-gradient(           to right,           #FFDD83,           #98DFD6         );
  }
}

@media (width >= 768px) {
  .c23 {
    border-bottom: 1px solid #98DFD6;
    border-right: none;
  }
}

@media (width >= 768px) {
  .c9 {
    width: 100%;
    height: 100%;
    margin-bottom: 0px;
    position: unset;
    top: unset;
    left: unset;
    rotate: 0deg;
    row-gap: 10px;
  }
}

@media (width >= 768px) {
  .c18 {
    font-family: fontFamily;
    font-weight: 400;
    color: #FBFBFB;
    font-size: 1.5rem;
    line-height: normal;
  }
}

@media (width >= 768px) {
  .c20 {
    font-family: fontFamily;
    font-weight: 400;
    color: #FBFBFB;
    font-size: 0.9rem;
    line-height: normal;
  }
}

@media (width >= 768px) {
  .c21 {
    margin-right: 8px;
  }

  .c21 svg {
    width: 16px;
    height: 16px;
  }
}

@media (width >= 768px) {
  .c10 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c11 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c12 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c13 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c14 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c15 {
    width: 140px;
    height: 140px;
  }
}

@media (width >= 768px) {
  .c16 {
    width: 140px;
    height: 140px;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      />
      <div
        class="c2"
      >
        <div
          class="c3"
          role="presentation"
        >
          <div
            class="c4"
          >
            <svg
              class="c5"
              color="#FFDD83"
            >
              portal.svg
            </svg>
          </div>
          <div
            class="c6"
          >
            <svg
              class="c5"
              color="#98DFD6"
            >
              portal.svg
            </svg>
          </div>
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <div
                class="c9"
                data-testid="contact-info"
              >
                <div
                  class="c10"
                >
                  <img
                    alt="My profile picture 1"
                    class="c11"
                    data-nimg="1"
                    decoding="async"
                    height="140"
                    loading="lazy"
                    src="/_next/image?url=%2Fassets%2Fportrait-1.png&w=384&q=75"
                    srcset="/_next/image?url=%2Fassets%2Fportrait-1.png&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait-1.png&w=384&q=75 2x"
                    style="color: transparent;"
                    width="140"
                  />
                  <img
                    alt="My profile picture 2"
                    class="c12"
                    data-nimg="1"
                    decoding="async"
                    height="140"
                    loading="lazy"
                    src="/_next/image?url=%2Fassets%2Fportrait-2.png&w=384&q=75"
                    srcset="/_next/image?url=%2Fassets%2Fportrait-2.png&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait-2.png&w=384&q=75 2x"
                    style="color: transparent;"
                    width="140"
                  />
                  <img
                    alt="My profile picture 3"
                    class="c13"
                    data-nimg="1"
                    decoding="async"
                    height="140"
                    loading="lazy"
                    src="/_next/image?url=%2Fassets%2Fportrait-3.png&w=384&q=75"
                    srcset="/_next/image?url=%2Fassets%2Fportrait-3.png&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait-3.png&w=384&q=75 2x"
                    style="color: transparent;"
                    width="140"
                  />
                  <img
                    alt="My profile picture 4"
                    class="c14"
                    data-nimg="1"
                    decoding="async"
                    height="140"
                    loading="lazy"
                    src="/_next/image?url=%2Fassets%2Fportrait-4.png&w=384&q=75"
                    srcset="/_next/image?url=%2Fassets%2Fportrait-4.png&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait-4.png&w=384&q=75 2x"
                    style="color: transparent;"
                    width="140"
                  />
                  <img
                    alt="My profile picture 5"
                    class="c15"
                    data-nimg="1"
                    decoding="async"
                    height="140"
                    loading="lazy"
                    src="/_next/image?url=%2Fassets%2Fportrait-5.png&w=384&q=75"
                    srcset="/_next/image?url=%2Fassets%2Fportrait-5.png&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait-5.png&w=384&q=75 2x"
                    style="color: transparent;"
                    width="140"
                  />
                  <img
                    alt="My profile picture 6"
                    class="c16"
                    data-nimg="1"
                    decoding="async"
                    height="140"
                    loading="lazy"
                    src="/_next/image?url=%2Fassets%2Fportrait-6.png&w=384&q=75"
                    srcset="/_next/image?url=%2Fassets%2Fportrait-6.png&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fportrait-6.png&w=384&q=75 2x"
                    style="color: transparent;"
                    width="140"
                  />
                </div>
                <h3
                  class="c17 c18"
                >
                  Software Engineer
                </h3>
                <p
                  class="c19 c20"
                >
                  <span
                    class="c21"
                  >
                    <svg>
                      linkedIn.svg
                    </svg>
                  </span>
                  <a
                    aria-label="My LinkedIn profile"
                    class="c22 "
                    data-testid="plain-anchor"
                    href="https://www.linkedin.com/in/sauldeleonguerrero"
                    rel="noreferrer"
                    target="_blank"
                  >
                    LinkedIn
                  </a>
                </p>
                <p
                  class="c19 c20"
                >
                  <span
                    class="c21"
                  >
                    <svg>
                      email.svg
                    </svg>
                  </span>
                  <a
                    aria-label="My electronic mail address"
                    class="c22 "
                    data-testid="plain-anchor"
                    href="mailto:sauldeleonguerrero@gmail.com"
                    rel="noreferrer"
                    target="_blank"
                  >
                    sauldeleonguerrero@gmail.com
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c23"
      />
    </div>
  </div>
</body>
`;
